text_data.py

```diff
@@ -77,13 +77,16 @@ def encode_truncated_ds(ds):
         
         def collate_batch(batch):
             max = 0
+            total_tokens = 0
             out_batch = []
+            num_tokens = []
             for l in batch['input_ids']:
+                num_tokens += [len(l)]
                 if len(l) > max:
                     max = len(l)
                 out_batch.append(torch.tensor(l, dtype=torch.long))
             seq = pad_sequence(out_batch, batch_first=True, padding_value=_tokenizer().pad_token_id)
-            return { 'input_ids': seq }
+            return { 'input_ids': seq, 'num_tokens': num_tokens }
```

Comment:
The columns here need to be iterable, so it won't do to actually sum the tokens here. We keep them separate like the different data items.